src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/GVMDependenciesNode.cs

```diff
@@ -130,13 +137,11 @@ public override IEnumerable<CombinedDependencyListEntry> SearchDynamicDependenci
                 if (!potentialOverrideType.IsDefType)
                     continue;
 
-                Debug.Assert(!potentialOverrideType.IsRuntimeDeterminedSubtype);
```

Comment:
Deleting this because IsRuntimeDeterminedSubtype is expensive and getting one here would mean we already have a bad EEType in the system. EETypeNode asserts this is not the case.